test/show.jl

```diff
@@ -70,14 +70,29 @@ end
 @testset "collection of colors" begin
     iob = IOBuffer()
     summary(iob, Gray{N0f8}[0.2, 0.4, 0.6])
-    @test String(take!(iob)) == "3-element Array{Gray{N0f8},1} with eltype Gray{Normed{UInt8,8}}"
+    vec_summary = String(take!(iob))
+
     mat = RGB{Float64}[RGB(1,0,0) RGB(0,1,0); RGB(0,0,1) RGB(1,1,1)]
     summary(iob, mat)
-    @test String(take!(iob)) == "2×2 Array{RGB{Float64},2} with eltype RGB{Float64}"
+    mat_summary = String(take!(iob))
+
     summary(iob, view(mat,:,:))
-    @test String(take!(iob)) == "2×2 view(::Array{RGB{Float64},2}, :, :) with eltype RGB{Float64}"
+    view_summary = String(take!(iob))
+
     summary(iob, TransparentColor[ARGB32(), HSVA(30,1,1,0.5)])
-    @test String(take!(iob)) == "2-element Array{TransparentColor,1} with eltype TransparentColor"
+    avec_summary = String(take!(iob))
+
+    if VERSION >= v"1.6.0-DEV.356"
+        @test_broken vec_summary == "3-element Vector{Gray{N0f8}}"
+        @test_broken mat_summary == "2×2 Matrix{RGB{Float64}}"
+        @test_broken view_summary == "2×2 view(::Matrix{RGB{Float64}}, :, :) with eltype RGB{Float64}"
+        @test_broken avec_summary == "2-element Vector{TransparentColor}"
+    else
+        @test vec_summary == "3-element Array{Gray{N0f8},1} with eltype Gray{Normed{UInt8,8}}"
```

Comment:
We can use `typeof(a)` here instead to avoid this version check.

See also: https://github.com/JuliaArrays/OffsetArrays.jl/pull/125

Reply:
That's right, but that looks like a replica of the implementation. :confused:

Reply:
I don't see if there is any issue as long as it is consistent with the Base implementation.

Reply:
Seems not a problem to me. It doesn't reuse any show methods defined in this package.

Reply:
These are the tests for `showarg` rather than `summary`.
My concern is the quality of the test, not the implementation. There are two different points of view:
- we have succeeded in detecting the specification change
- we have failed to follow the specification change

Reply:
There's no doubt that it gives better test quality to track every specific change. However, for IO changes, IMO it's really exhausting to track down every detail. So as a tradeoff between maintainability and test quality, I feel it okay as long as it follows a same pattern to Base.

Of course, this is just my preference.

Reply:
Note that there is no "with eltype" for `Array` in `Base`. "with eltype N0f8" is not wrong, but it is not the expected result.

Reply:
I think you might misunderstand my solution:

```julia
iob = IOBuffer()
avec = Gray{N0f8}[0.2, 0.4, 0.6]
vec_summary = summary(iob, avec)

...

@test vec_summary == "3-element $(typeof(avec)) with eltype Gray{Normed{UInt8,8}}"
```

Reply:
I'm sorry, I was assuming you were suggesting `eltype with $(eltype(avec))`.
If so, what are you going to test? :joy: